### ISSUE TITLE
Focusing an obscured text field on Android will silence TalkBack

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#21030]
To reproduce, run the gallery with TalkBack enabled and navigate to the text fields demo. Select the password text field and then hit back.  TalkBack will no longer announce any labels when elements are focused.

I've confirmed this still occurs 16 days ago.

**Comment by jonahwilliams** [comment]
I believe the most likely cause is that we are failing to send a `TYPE_VIEW_ACCESSIBILITY_FOCUS_CLEARED`, leading to TalkBack thinking we are still editing an obscured text field.

**Comment by jonahwilliams** [comment]
The above is not the issue. This does not happen if TalkBack is configured to speak passwords ( or if you are wearing headphones )

**Comment by jonahwilliams** [comment]
The problem can be seen here, in the tail of the TalkBack logs emitted when focusing a labeled element:

```
08-26 14:15:48.960 32653 32653 V ParseTree:                                     switch (#VERBOSITY_DESCRIPTION_ORDER):
08-26 14:15:48.960 32653 32653 V ParseTree:                                     case StateNameRolePosition:
08-26 14:15:48.960 32653 32653 V ParseTree:                                       %node_text_and_role
08-26 14:15:48.960 32653 32653 V ParseTree:                                         %get_node_text
08-26 14:15:48.960 32653 32653 V ParseTree:                                           fallback (2 items):
08-26 14:15:48.960 32653 32653 V ParseTree:                                             if ($global.lastTextEditIsPassword && !$global.speakPasswordsServicePolicy && ($event.sourceIsKeyboard || $node.isPinKey)):
08-26 14:15:48.960 32653 32653 V ParseTree:                                               Evaluating: $global.lastTextEditIsPassword && !$global.speakPasswordsServicePolicy && ($event.sourceIsKeyboard || $node.isPinKey)
08-26 14:15:48.960 32653 32653 V ParseTree:                                               ParseTreeVariableNode.resolveToBoolean() name=global.lastTextEditIsPassword value=true
08-26 14:15:48.960 32653 32653 V ParseTree:                                               ParseTreeVariableNode.resolveToBoolean() name=global.speakPasswordsServicePolicy value=false
08-26 14:15:48.960 32653 32653 V ParseTree:                                               ParseTreeVariableNode.resolveToBoolean() name=event.sourceIsKeyboard value=false
```

**Comment by jonahwilliams** [comment]
And here is an example of it working properly:

```
08-26 14:58:10.237  3333  3333 V ParseTree:                                     switch (#VERBOSITY_DESCRIPTION_ORDER):
08-26 14:58:10.237  3333  3333 V ParseTree:                                     case StateNameRolePosition:
08-26 14:58:10.237  3333  3333 V ParseTree:                                       %node_text_and_role
08-26 14:58:10.237  3333  3333 V ParseTree:                                         %get_node_text
08-26 14:58:10.237  3333  3333 V ParseTree:                                           fallback (2 items):
08-26 14:58:10.237  3333  3333 V ParseTree:                                             if ($global.lastTextEditIsPassword && !$global.speakPasswordsServicePolicy && ($event.sourceIsKeyboard || $node.isPinKey)):
08-26 14:58:10.237  3333  3333 V ParseTree:                                               Evaluating: $global.lastTextEditIsPassword && !$global.speakPasswordsServicePolicy && ($event.sourceIsKeyboard || $node.isPinKey)
08-26 14:58:10.237  3333  3333 V ParseTree:                                               ParseTreeVariableNode.resolveToBoolean() name=global.lastTextEditIsPassword value=false
08-26 14:58:10.237  3333  3333 V ParseTree:                                             if (false)
08-26 14:58:10.237  3333  3333 V ParseTree:                                               Evaluating: $node.contentDescription
08-26 14:58:10.237  3333  3333 V ParseTree:                                               ParseTreeVariableNode.resolveToString() name=node.contentDescription value=
08-26 14:58:10.237  3333  3333 V ParseTree:                                             Evaluating: $node.text
08-26 14:58:10.238  3333  3333 V ParseTree:                                             ParseTreeVariableNode.resolveToString() name=node.text value=B
08-26 14:58:10.238  3333  3333 V ParseTree: This item represents B.
08-26 14:58:10.238  3333  3333 V ParseTree: Even more additional list item information appears on line three.
```
